### PR TITLE
[NO GBP] Emote Panel TGUI, now should be fixed

### PR DESCRIPTION
**Pull request by larentoun**
## About The Pull Request
Original: https://github.com/tgstation/tgstation/pull/79434
[Removed](https://github.com/tgstation/tgstation/commit/884121b35c12cd73bf61cb0b2d9554960908222f) new() from TGUI input, it uses keys instead + checks if the key is in GLOB.emote_list

Below is copypaste from the original post but with updated images:

Adds Emote Panel TGUI in IC category. It shows all the available emotes for the character on the moment of opening the panel. It also has filters for:
* If the emote is visible;
* If the emote is audible;
* If the emote has sound;
* If the emote requires hands;
* If the emote can have params (such as target);

<details><summary>Images</summary>

Screenshot_1
Screenshot_2
Screenshot_3
Screenshot_4
Screenshot_5
Screenshot_6
Screenshot_7

</details>

## Why It's Good For The Game
Easier for newer players to use emotes without the use of "*help" and keybindings. Also shows if the emote is visible/audible/etc, so there is less need to spam each of them to see which have sound.

## Changelog
:cl:
add: Emote Panel TGUI added in IC category.
/:cl:

### CODE REVIEW

File: code/modules/emote_panel/emote_panel.dm [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		if("play_emote")
+			var/emote_key = params["emote_key"]
+			if(isnull(emote_key) || !GLOB.emote_list[emote_key])
+				return
+			var/use_params = params["use_params"]
+			var/datum/emote/emote = GLOB.emote_list[emote_key][1]
+			var/emote_param
+			if(emote.message_param && use_params)
+				emote_param = tgui_input_text(usr, "Add params to the emote...", emote.message_param)
+			usr.emote(emote_key, message = emote_param, intentional = TRUE)
```
Comment (ZephyrTFA) [inline code/modules/emote_panel/emote_panel.dm:46]: don't use usr
ui.user is what you want

File: code/modules/emote_panel/emote_panel.dm [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		ui.open()
+
+/datum/emote_panel/ui_state(mob/user)
+	return GLOB.always_state
+
+/mob/living/verb/emote_panel()
+	set name = "Emote Panel"
+	set category = "IC"
+
+	var/static/datum/emote_panel/emote_panel = new
```
Comment (ZephyrTFA) [inline code/modules/emote_panel/emote_panel.dm:61]: use lazy initialization for static vars which aren't a base type

Comment (larentoun) [reply code/modules/emote_panel/emote_panel.dm:61]: Can you elaborate on this one? LAZY seems to be used only on lists if I'm not mistaken?

Comment (ZephyrTFA) [reply code/modules/emote_panel/emote_panel.dm:61]: don't have a static var, which isnt a string or float, have an initializer like that.
You'll want to do something such as
```
var/static/your/type/here/static_var
if(isnull(static_var))
	static_var = default value
```

Comment (ZephyrTFA) [reply code/modules/emote_panel/emote_panel.dm:61]: It's an incredibly minor nit, and this is not blocking.
I'd just be happy if you did it \:)

File: code/modules/emote_panel/emote_panel.dm [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		if("play_emote")
+			var/emote_key = params["emote_key"]
+			if(isnull(emote_key) || !GLOB.emote_list[emote_key])
+				return
+			var/use_params = params["use_params"]
+			var/datum/emote/emote = GLOB.emote_list[emote_key][1]
+			var/emote_param
+			if(emote.message_param && use_params)
+				emote_param = tgui_input_text(ui.user, "Add params to the emote...", emote.message_param)
+			ui.user.emote(emote_key, message = emote_param, intentional = TRUE)
```
Comment (JohnFulpWillard) [inline code/modules/emote_panel/emote_panel.dm:46]: You should check for ``emote_param`` being null probably.

Comment (larentoun) [reply code/modules/emote_panel/emote_panel.dm:46]: Sending null is fine since it's then checked down below in the code for `emote()`

**Pull request merged**